Apps/W1/Shopify/app/src/Payments/Codeunits/ShpfyPaymentTermsAPI.Codeunit.al

```diff
@@ -64,9 +64,22 @@ codeunit 30168 "Shpfy Payment Terms API"
         JsonHelper.GetValueIntoField(JTemplate, 'description', PaymentTermRecordRef, ShpfyPaymentTerms.FieldNo(Description));
         PaymentTermRecordRef.SetTable(ShpfyPaymentTerms);
 
+        if ShpfyPaymentTerms.Type = 'FIXED' then
+            if ShouldBeMarkedAsPrimary() then
+                ShpfyPaymentTerms.Validate("Is Primary", true);
+
         if IsNew then
             ShpfyPaymentTerms.Insert(true)
         else
             ShpfyPaymentTerms.Modify(true);
     end;
+
+    local procedure ShouldBeMarkedAsPrimary(): Boolean
```

Comment:
What about a situation when I haven’t synced payment terms at all yet? 
will it still be a silent success because of that?

Reply:
No, if no payment terms are mapped or marked as Is Primary, then the invoice is marked -2 

![image](https://github.com/tinestaric/ALAppExtensions/assets/56364962/a9fb17d6-2086-4ff4-abd7-5750c77c84e4)